Bottom: 0f2efee48a61eb345a7f7dd4ebbf0df79e739ea9
Top:    bc9105b8cf4060b87bc61c07dc49b43af079589c
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2013-06-10 10:57:56 -0700

init: Remove USE_IB_ACM configuration option

When the librdmacm is configured, it sets the USE_IB_ACM option
if infininband/acm.h is found.  We can remove this option with
very little overhead, which would allow a user to install
ACM after installing the librdmacm, and the librdmacm would be
able to make use of ACM.

Signed-off-by: Sean Hefty <sean.hefty@intel.com>



---

diff --git a/configure.ac b/configure.ac
index 31b1a0e..6163f56 100644
--- a/configure.ac
+++ b/configure.ac
@@ -70,9 +70,6 @@ AC_CHECK_HEADER(valgrind/memcheck.h, [],
 fi
 fi
 
-AC_CHECK_HEADER(infiniband/acm.h, 
-	AC_DEFINE([USE_IB_ACM], 1, [setting IBACM support]), [])
-
 AC_CHECK_HEADER(infiniband/acm.h,
 	AC_CHECK_MEMBER(struct acm_msg.resolve_data, [],
 		AC_DEFINE(DEFINE_ACM_MSG, 1, [adding ACM message definition]),
diff --git a/src/acm.c b/src/acm.c
index 6e8e173..dfc4959 100644
--- a/src/acm.c
+++ b/src/acm.c
@@ -45,7 +45,6 @@
 #include <infiniband/ib.h>
 #include <infiniband/sa.h>
 
-#ifdef USE_IB_ACM
 #include <infiniband/acm.h>
 
 #if DEFINE_ACM_MSG
@@ -401,5 +400,3 @@ void ucma_ib_resolve(struct rdma_addrinfo **rai, struct rdma_addrinfo *hints)
 	if (af_ib_support && !(hints->ai_flags & RAI_ROUTEONLY) && (*rai)->ai_route_len)
 		ucma_resolve_af_ib(rai);
 }
-
-#endif /* USE_IB_ACM */
diff --git a/src/cma.h b/src/cma.h
index b575b9b..e6fba8b 100644
--- a/src/cma.h
+++ b/src/cma.h
@@ -162,15 +162,9 @@ extern int af_ib_support;
 
 #define RAI_ROUTEONLY		0x01000000
 
-#ifdef USE_IB_ACM
 void ucma_ib_init();
 void ucma_ib_cleanup();
 void ucma_ib_resolve(struct rdma_addrinfo **rai, struct rdma_addrinfo *hints);
-#else
-#define ucma_ib_init()
-#define ucma_ib_cleanup()
-#define ucma_ib_resolve(x, y)
-#endif
 
 /* Define path record definition if using older version of libibverbs */
 #ifdef DEFINE_PATH_RECORD
